compiler/src/main/java/io/neow3j/compiler/converters/ObjectsConverter.java

```diff
@@ -92,6 +100,67 @@ public AbstractInsnNode convert(AbstractInsnNode insn, NeoMethod neoMethod, Comp
         return insn;
     }
 
+    private AbstractInsnNode handleContractInterfaceCtor(AbstractInsnNode insn, NeoMethod neoMethod,
+            CompilationUnit compUnit) throws IOException {
+
+        TypeInsnNode typeInsn = (TypeInsnNode) insn;
+        assert typeInsn.getNext().getOpcode() == JVMOpcode.DUP.getOpcode() :
+                "Expected DUP after NEW but got other instructions";
+
+        ClassNode ownerClassNode = getAsmClassForInternalName(typeInsn.desc, compUnit.getClassLoader());
+        insn = typeInsn.getNext().getNext();
+        while (!isCallToCtor(insn, ownerClassNode.name)) {
+            insn = handleInsn(insn, neoMethod, compUnit);
+            insn = insn.getNext();
+        }
+        MethodNode methodNode = getMethodNode((MethodInsnNode) insn, ownerClassNode).get();
+        Type[] cTorArgTypes = Type.getType(methodNode.desc).getArgumentTypes();
+        int cTorParamLength = cTorArgTypes.length;
+        if (hasAnnotations(ownerClassNode, NativeContract.class)) {
+            return handleNativeContractHash(neoMethod, ownerClassNode, insn);
+        }
+        if (cTorParamLength == 1 && getInternalName(Hash160.class).equals(cTorArgTypes[0].getInternalName())) {
+            // The instructions that lead to the contract hash are already on the stack at this point.
+            return insn;
+        }
+        throw new CompilerException(
+                format("A constructor of a ContractInterface is required to take exactly one %s type as parameter.",
+                        getInternalName(Hash160.class)));
+    }
+
+    private AbstractInsnNode handleNativeContractHash(NeoMethod neoMethod, ClassNode ownerClassNode,
+            AbstractInsnNode insn) {
+
+        AnnotationNode annotation = AsmHelper.getAnnotationNode(ownerClassNode, NativeContract.class).get();
+        io.neow3j.types.Hash160 scriptHash;
+        try {
+            scriptHash = new io.neow3j.types.Hash160((String) annotation.values.get(1));
+        } catch (IllegalArgumentException e) {
+            throw new CompilerException(ownerClassNode,
+                    format("Script hash '%s' of the contract class '%s' does not have the length of a valid script " +
+                            "hash.", annotation.values.get(1), getClassNameForInternalName(ownerClassNode.name)));
+        }
+        neoMethod.addInstruction(buildPushDataInsn(reverseArray(scriptHash.toArray())));
+        while (!isCallToCtor(insn, ownerClassNode.name)) {
+            insn = insn.getNext();
+        }
```

Comment:
Is this loop necessary? In my understanding, we are already at the constructor call instruction when we are handling the native contract hash (because of the loop at line 112).

Reply:
You are right, I'll remove that loop.